### PR TITLE
Update to new pixel logos and improve macOS UI logos

### PR DESCRIPTION
**Pull request by brandelune**
New 16x16 and 32x32 pixels logos based on the "modern" new logo.

### CODE REVIEW

**Comment by brandelune** [comment]
I'll modify the scripts later to complete the PR so that the png generate the gif that we use.

**Comment by amake** [comment]
It would help me move forward with this PR if it could describe (ideally with screenshots) where all the intended changed icons are. Currently I'm not sure exactly where we're changing and where we're not.

> There must be a way to have Java reduce the display size of bigger icons so that the pixelization is not that bad

Can you provide a screenshot showing where the pixelation is bad?

**Comment by brandelune** [comment]
I intended to change the OmegaT icon that's displayed from OmegaT's file chooser when an OmegaT project is detected in the chooser.

Screen Shot 2022-03-19 at 23 03 49

I found that the icon I had had created was too pixelized and thought that maybe having Java use the already existing "modern" icons and just reduce their size would be the best option to have a nice modern OmegaT icon displayed in the chooser.

**Comment by miurahr** [comment]
Here is an example of changes
jc-logo1

**Comment by brandelune** [comment]
I fear that the icon is a bit too pixelized to be of any use though.

**Comment by miurahr** [comment]
I think this is worth merging. I don't find any obvious problem when using.

**Comment by brandelune** [comment]
No. The icon is too pixelized. I need to find the right size for the icon. Give me some more time.

**Comment by miurahr** [comment]
@brandelune progress?

**Comment by brandelune** [comment]
I'll work on that when I'm done with the manual.

**Comment by brandelune** [comment]
Screen Shot 2022-08-23 at 21 01 22

This is what I have on macOS. The OmegaT icon looks much more pixelized than the other icons. I don't think it is acceptable.

I guess swing has a way to downsize macOS icons so that they look good, but since I guess it's not exactly the process that we use here (we don't show an existing OS icon, we overlay our icon), I guess we don't have access to that "feature" (if there really is one)...

**Comment by miurahr** [comment]
> Screen Shot 2022-08-23 at 21 01 22
> 
> This is what I have on macOS. The OmegaT icon looks much more pixelized than the other icons. I don't think it is acceptable.
> 
> I guess swing has a way to downsize macOS icons so that they look good, but since I guess it's not exactly the process that we use here (we don't show an existing OS icon, we overlay our icon), I guess we don't have access to that "feature" (if there really is one)...

macOS and other modern OSes may ask applications to provide more icon sizes for iconset.

OmegaT provide small subset of it.

- [x] 16x16
- [ ] 16x16@2x  for retina
- [x] 32x32
- [ ] 32x32@2x
- [ ] 64x64
- [ ] 64x64@2x
- [ ] 128x128
- [ ] 128x128@2x
- [ ] 256x256
- [ ] 256x256@2x
- [ ] 512x512
- [ ] 512x512@2x
- [x] Scalable(SVG)

ref https://gist.github.com/jamieweavis/b4c394607641e1280d447deed5fc85fc

**Comment by brandelune** [comment]
Thank you for looking into that. Is there a way to fix the version that OmegaT provides ?

**Comment by miurahr** [comment]
Please run `images/make_osx_icon.sh` to create iconset. If it is not enough for recent version of macOS, we need to update script.

**Comment by miurahr** [comment]
> Please run `images/make_osx_icon.sh` to create iconset. If it is not enough for recent version of macOS, we need to update script.

Actually not enough. It create  16x16 32x32 32x32@2x  128x128 and 256x256

- 512px version should be taken from website:  http://briac.net/omegat-logo/

There we may need 16x16@2x, 64x64 64x64@2x, 128x128@2x, 256x256@2x and 512x512 512x512@2x

**Comment by miurahr** [comment]
I've force pushed to your repository with update of the script.
Please force pull again the code. @brandelune

**Comment by brandelune** [comment]
Ok. I will.

**Comment by brandelune** [comment]
done.

**Comment by miurahr** [comment]
`icon_1024x1024.png` should be `icon_512x512@2x.png` and there is no `icon_16x16.png`?

**Comment by brandelune** [comment]
```
➜  images git:(topic/jc/pixelLogos) ✗ ./make_osx_icon.sh
Warning: Option --export-png= is deprecated
Warning: Option --file= is deprecated
Fontconfig warning: ignoring UTF-8: not a valid region tag
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_32x32.png
   Total length of data found in critical chunks            =      2014
   Best pngcrush method        =   7 (ws 13 fm 0 zl 9 zs 0) =      1973
CPU time decode 0.000587, encode 0.001313, other 0.001515, total 0.003453 sec
Warning: Option --export-png= is deprecated
Warning: Option --file= is deprecated
Fontconfig warning: ignoring UTF-8: not a valid region tag
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_64x64.png
   Total length of data found in critical chunks            =      5095
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =      5086
CPU time decode 0.001353, encode 0.002850, other 0.001309, total 0.005595 sec
Warning: Option --export-png= is deprecated
Warning: Option --file= is deprecated
Fontconfig warning: ignoring UTF-8: not a valid region tag
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_128x128.png
   Total length of data found in critical chunks            =     13056
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =     12089
CPU time decode 0.003377, encode 0.012277, other 0.001761, total 0.017612 sec
Warning: Option --export-png= is deprecated
Warning: Option --file= is deprecated
Fontconfig warning: ignoring UTF-8: not a valid region tag
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_256x256.png
   Total length of data found in critical chunks            =     32881
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =     28790
CPU time decode 0.008061, encode 0.040941, other 0.001542, total 0.050901 sec
Warning: Option --export-png= is deprecated
Warning: Option --file= is deprecated
Fontconfig warning: ignoring UTF-8: not a valid region tag
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_1024x1024.png
   Total length of data found in critical chunks            =    218630
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =    169455
CPU time decode 0.089563, encode 0.608600, other 0.002490, total 0.702002 sec
cp: /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_512x512.png: No such file or directory
```

**Comment by miurahr** [comment]
From shell comment

>  512px version is taken from website:
>  http://briac.net/omegat-logo/
>

**Comment by miurahr** [comment]
Update script  to generate 512 px version also and fix script error that 512px version is not exist.

**Comment by brandelune** [comment]
Things work now.

**Comment by miurahr** [comment]
If UI on macOS is ok, I'd like to squash and merge it.

File: images/OmegaT pixels/contributor.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Leonardo Kenji Itiama Fulukava
```
Comment (miurahr) [inline images/OmegaT pixels/contributor.txt:1]: Path `OmegaT pixels` has a space character. In order to decrease trouble, Could you mind replace space chr with `_` or '.'?

**Comment by brandelune** [comment]
Screen Shot 2022-09-28 at 18 36 16
Screen Shot 2022-09-28 at 18 35 47

Screen Shot 2022-09-28 at 18 38 26
Screen Shot 2022-09-28 at 18 38 13

The icons are replaced. But the result it pixelized and not pretty.

The quality that we should have in the About dialog is icon_64x64@2x.png, and the one that we should have in the open project is icon_32x32@2x.png.

**Comment by miurahr** [comment]
I now recognize About dialog uses very small resolution of icon image.
It should be fixed in another activity. 
Here we can focus on macOS issues.

**Comment by miurahr** [comment]
https://en.wikipedia.org/wiki/Apple_Icon_Image_format

> The Apple Icon Image format is an [icon](https://en.wikipedia.org/wiki/Icon_(computing)) format used in [Apple Inc.](https://en.wikipedia.org/wiki/Apple_Inc.)'s [macOS](https://en.wikipedia.org/wiki/MacOS). It supports icons of 16 × 16, 32 × 32, 48 × 48, 128 × 128, 256 × 256, 512 × 512 points at 1x and 2x scale, with both [1-](https://en.wikipedia.org/wiki/Bit) and [8-bit](https://en.wikipedia.org/wiki/8-bit) [alpha channels](https://en.wikipedia.org/wiki/Alpha_compositing) and multiple image states (example: open and closed folders). The fixed-size icons can be scaled by the operating system and displayed at any intermediate size.

> As of [macOS 11](https://en.wikipedia.org/wiki/MacOS_11), asset catalogs are the preferred file format for macOS custom icons instead.[[1]](https://en.wikipedia.org/wiki/Apple_Icon_Image_format#cite_note-1)

Should we move to asset catalogs? what is it?

File: images/make_osx_icon.sh [hunk 2]
```diff
@@ -13,23 +13,26 @@ PNGCRUSH="$(which pngcrush)"
 SVG_FILE="${SCRIPT_ROOT}/OmegaT.svg"
 [ ! -f "$SVG_FILE" ] && echo "Can't find SVG to convert." && exit 1
 
-SIZES="32 64 128 256"
-# 512px version is taken from website:
-# http://briac.net/omegat-logo/
+SIZES="32 64 128 256 512 1024"
```
Comment (miurahr) [inline images/make_osx_icon.sh:16]: Is `icon_16x16.png` generated?

**Comment by miurahr** [comment]
> New 16x16 and 32x32 pixels logos based on the "modern" new logo.

Who use these `images/OmegaT_pixels/32x32/*` images? I cannot find any code to refer these.

**Comment by brandelune** [comment]
I think they are converted into 

    public static final Image APP_ICON_32X32 = getBundledImage("OmegaT.gif");
    public static final Image APP_ICON_16X16 = getBundledImage("OmegaT_small.gif");


See:
/src/org/omegat/util/gui/ResourcesUtil.java

File: images/make_osx_icon.sh (hunk 2 above)
Comment (brandelune) [reply images/make_osx_icon.sh]: I just did and pushed it. Originally, I had one created by a pixel artist (the one that's in the repository). But it was too pixelized.

**Comment by brandelune** [comment]
> > As of [macOS 11](https://en.wikipedia.org/wiki/MacOS_11), asset catalogs are the preferred file format for macOS custom icons instead.[[1]](https://en.wikipedia.org/wiki/Apple_Icon_Image_format#cite_note-1)
> 
> Should we move to [asset catalogs](https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_ref-Asset_Catalog_Format/AppIconType.html)?

No because macOS 11 is too recent.

**Comment by miurahr** [comment]
Only a asset catalogs can have @3x scaling logos, which is required for iMac 4k, 5k

**Comment by miurahr** [comment]
> It supports icons of 16 × 16, 32 × 32, 48 × 48, 128 × 128, 256 × 256, 512 × 512 points at 1x and 2x scale

There is no 64x64 and 64x64@2x.

File: images/make_osx_icon.sh [hunk 3]
```diff
@@ -13,23 +13,26 @@ PNGCRUSH="$(which pngcrush)"
...
+
     pngcrush -ow "$OUTFILE"
 done
 
-# The system likes 64x64 to be named 32x32@2x for some reason.
-mv "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64.png" \
-   "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_16x16@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_128x128.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64@2x.png"
```
Comment (miurahr) [inline images/make_osx_icon.sh:32]:
```suggestion
mv "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32@2x.png"
```

File: images/make_osx_icon.sh [hunk 4]
```diff
@@ -13,23 +13,26 @@ PNGCRUSH="$(which pngcrush)"
 SVG_FILE="${SCRIPT_ROOT}/OmegaT.svg"
 [ ! -f "$SVG_FILE" ] && echo "Can't find SVG to convert." && exit 1
 
-SIZES="32 64 128 256"
-# 512px version is taken from website:
-# http://briac.net/omegat-logo/
+SIZES="16 32 64 128 256 512 1024"
```
Comment (miurahr) [inline images/make_osx_icon.sh:16]:
```suggestion
SIZES="16 32 48 64 128 256 512 1024"
```

**Comment by miurahr** [comment]
https://web.archive.org/web/20180618155438/https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/

Apple developer document suggest icons should be



Attribute | Value
-- | --
Format | PNG
Color space | sRGB
Layers | Flattened with transparency as appropriate.
Resolution | @1x and @2x. See Image Size and Resolution
Shape | Square canvas. Allow transparency to define the icon shape.

and
sizes are



Icon size (@1x) | Icon size (@2x)
-- | --
512px × 512px (512pt × 512pt @1x) | 1024px × 1024px (512pt × 512pt @2x)
256px × 256px (256pt × 256pt @1x) | 512px × 512px (256pt × 256pt @2x)
128px × 128px (128pt × 128pt @1x) | 256px × 256px (128pt × 128pt @2x)
32px × 32px (32pt × 32pt @1x) | 64px × 64px (32pt × 32pt @2x)
16px × 16px (16pt × 16pt @1x) | 32px × 32px (16pt × 16pt @2x)

**Comment by miurahr** [comment]
From wikipedia table
https://en.wikipedia.org/wiki/Apple_Icon_Image_format




OSType | Length (bytes) | Size (pixels) | Supported OS Version | Description
-- | -- | -- | -- | --
icp4 | varies | 16x16 | 10.7 | JPEG 2000† or PNG† format or 24-bit RGB icon[2]
icp5 | varies | 32x32 | 10.7 | JPEG 2000† or PNG† format or 24-bit RGB icon[2]
icp6 | varies | 48x48 | 10.7 | JPEG 2000† or PNG† format
ic07 | varies | 128x128 | 10.7 | JPEG 2000 or PNG format
ic08 | varies | 256x256 | 10.5 | JPEG 2000 or PNG format
ic09 | varies | 512x512 | 10.5 | JPEG 2000 or PNG format
ic10 | varies | 1024x1024 | 10.7 | JPEG 2000 or PNG format (512x512@2x "retina" in 10.8)
ic11 | varies | 32x32 | 10.8 | JPEG 2000 or PNG format (16x16@2x "retina")
ic12 | varies | 64x64 | 10.8 | JPEG 2000 or PNG format (32x32@2x "retina")
ic13 | varies | 256x256 | 10.8 | JPEG 2000 or PNG format (128x128@2x "retina")
ic14 | varies | 512x512 | 10.8 | JPEG 2000 or PNG format (256x256@2x "retina")

There is no `48x48@2x` size.

File: images/make_osx_icon.sh [hunk 5]
```diff
@@ -13,23 +13,26 @@ PNGCRUSH="$(which pngcrush)"
...
-mv "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64.png" \
-   "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_16x16@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_32x32@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_128x128.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_64x64@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_256x256.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_128x128@2x.png"
+cp "${SCRIPT_ROOT}/OmegaT.iconset/icon_512x512.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_256x256@2x.png"
+mv "${SCRIPT_ROOT}/OmegaT.iconset/icon_1024x1024.png" "${SCRIPT_ROOT}/OmegaT.iconset/icon_512x512@2x.png"
 
 iconutil --convert icns "${SCRIPT_ROOT}/OmegaT.iconset"
```
Comment (miurahr) [inline images/make_osx_icon.sh:37]:
```suggestion
iconutil --convert icns "${SCRIPT_ROOT}/OmegaT.iconset"
# install icnsutil
python -m venv icnsenv
source icnsenv/bin/activate
python -m pip install icnsutil
# print produced icons
icnsutil p "${SCRIPT_ROOT}/OmegaT.icns"

# verify valid format?
icnsutil t "${SCRIPT_ROOT}/OmegaT.icns"

```

Can we use [icnsutils](https://pypi.org/project/icnsutil/) to validate resulted `*.icns` file?

Comment (brandelune) [reply images/make_osx_icon.sh:37]: With your suggestion, here is what I have:

```
 ./make_osx_icon.sh                                                                                            
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_16x16.png
   Total length of data found in critical chunks            =       824
   Best pngcrush method        =   7 (ws 11 fm 0 zl 9 zs 0) =       823
CPU time decode 0.000207, encode 0.000552, other 0.001751, total 0.002529 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_32x32.png
   Total length of data found in critical chunks            =      2014
   Best pngcrush method        =   7 (ws 13 fm 0 zl 9 zs 0) =      1973
CPU time decode 0.000624, encode 0.001436, other 0.001554, total 0.003732 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_64x64.png
   Total length of data found in critical chunks            =      5095
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =      5086
CPU time decode 0.001336, encode 0.002847, other 0.001484, total 0.005752 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_128x128.png
   Total length of data found in critical chunks            =     13056
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =     12089
CPU time decode 0.003036, encode 0.009804, other 0.001405, total 0.014439 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_256x256.png
   Total length of data found in critical chunks            =     32881
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =     28790
CPU time decode 0.008542, encode 0.042574, other 0.001716, total 0.053281 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_512x512.png
   Total length of data found in critical chunks            =     83171
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =     68187
CPU time decode 0.025796, encode 0.174426, other 0.001915, total 0.202870 sec
  Recompressing IDAT chunks in /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.iconset/icon_1024x1024.png
   Total length of data found in critical chunks            =    218630
   Best pngcrush method        =   7 (ws 15 fm 0 zl 9 zs 0) =    169455
CPU time decode 0.091268, encode 0.621755, other 0.002331, total 0.716830 sec
Requirement already satisfied: icnsutil in ./icnsenv/lib/python3.10/site-packages (1.0.1)
File: /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.icns
  ic12: 5674 bytes, png: 32x32@2x
  ic07: 15278 bytes, png: 128x128
  ic13: 42071 bytes, png: 128x128@2x
  ic08: 42071 bytes, png: 256x256
  ic04: 869 bytes, argb: 16x16
  ic14: 117364 bytes, png: 256x256@2x
  ic09: 117364 bytes, png: 512x512
  ic05: 2650 bytes, argb: 32x32
  ic10: 343649 bytes, png: 512x512@2x
  ic11: 2190 bytes, png: 16x16@2x
  info: 310 bytes, plist: info

File: /Users/suzume/Documents/Repositories/omegat.brandelune/images/OmegaT.icns
OK
```

Comment (brandelune) [reply images/make_osx_icon.sh:37]: If that is what you expect, then let's go ahead. Do you need me to push anything ?

Comment (miurahr) [reply images/make_osx_icon.sh:37]: Oh,  it looks good.

Comment (miurahr) [reply images/make_osx_icon.sh:37]: We now have ic04, ic05, ic07, ic08, ic09, ic10, ic11, ic12, ic13, ic14 icons, 10 icon files.

I hope you can get clear view of icon on finder.

Comment (brandelune) [reply images/make_osx_icon.sh:37]: Screen Shot 2022-10-02 at 18 07 59

Comment (brandelune) [reply images/make_osx_icon.sh:37]: I don't see any difference. But if we agree that it's cute to see a pixelized OmegaT, then I'm fine with it. It's an improvement on having different logos anyway.

Comment (miurahr) [reply images/make_osx_icon.sh:37]: OK, I'd like to merge here.

**Comment by miurahr** [comment]
Squash and merged into `master`

**Pull request closed without merging**